Platform/CommonBoardPkg/Library/BoardSupportLib/BoardSupportLib.c

```diff
@@ -117,6 +118,9 @@ FillBootOptionListFromCfgData (
       // LBA should be defined as 64bit.
       // Will remove the typecast when the structure is fixed.
       BootOption->Image[ImageIdx].LbaImage.LbaAddr = (UINT32)Lba;
+    } else if ((StrPtr[0] == '0') && ((StrPtr[1] == 'x') || (StrPtr[1] == 'X')) &&
+               (AsciiStrHexToUint64S (StrPtr, NULL, &RamAddress) == RETURN_SUCCESS)) {
```

Comment:
It is okay to add a new format "0x<hex>" for BootImage. If so, you may also need to update Platform/CommonBoardPkg/CfgData/Template_BootOption.yaml to tell users that "0x<hex>" is for memory address.

Another idea is to reuse the format: "#0x<HEX>" ? (line 114 to 119 ?) According to my test, the format "#0x<hex>" already works for OsBootDeviceMemory, although its usage was designed for the for LBA  (logic block address) of block device. Maybe the only thing we need to do is to reword the term "LBA"  to a more generic term.

Reply:
Updated